Documentation/StringFormatting.md

```diff
@@ -0,0 +1,128 @@
+
+The alignment characters are:
+- `<`: Align left.
+- `>`: Align right.
+- `^`: Align centered.
+
+### Sign
+
+- `+`: Always display a sign before the number.
+- `-`: Display a sign for negative numbers only.
+- (space): Display a sign for negative numbers, and a leading space for other numbers.
+
+### Hash
+
+`#` causes an "alternate form" to be used.
+
+For integer types, this adds the number-base prefix after the sign:
+- `0b` for binary.
+- `0` for octal.
+- `0x` for hexadecimal.
+
+### Zero
+
+`0` pads the number with leading zeros.  
+
+### Width and Precision
+
+The width defines the minimum number of characters in the output. The precision is a `.` followed by a precision number, which is used as the precision of floating-point numbers, or a maximum-width for string values.
+
+Both the width and precision can be provided as a replacement field (`{}`, optionally including an argument index) which allows you to use an integer argument instead of a hard-coded number.
+
+### Type
+
+| Type     | Effect                | Example output           |
+|----------|-----------------------|--------------------------|
+| b        | binary                | `110`, `0b000110`        |
+| B        | binary uppercase      | `110`, `0B000110`        |
+| d        | decimal               | `42`, `+0000042`         |
+| o        | octal                 | `043`                    |
+| x        | hexadecimal           | `ff0`, `0x00000ff0`      |
+| X        | hexadecimal uppercase | `FF0`, `0X00000FF0`      |
+| c        | character             | `a`                      |
+| s        | string (default)      | `well, hello friends!`   |
+| p        | pointer               | `0xdeadc0de`             |
+| f        | float                 | `1.234`, `-inf`          |
+| a        | hex float             |                          |
+| A        | hex float uppercase   |                          |
+| hex-dump | hexadecimal dump      | `fdfdfdfd`, `3030    00` |
+
+Not all types are compatible with all input types, of course.
+
+`bool` values can be treated as an integer type (`0` or `1`), or as a string (`true` or `false`).
+
+`char` and `wchar` types can be treated as an integer type (their numeric value) or as a string.
+
+## Formatting custom types
+
+You can provide a custom `AK::Formatter<Foo>` class to format `Foo` values. For the simplest case where you already have a function that produces a string from your type, that would look like this:
+```c++
+
+namespace AK {
+template<>
+struct Formatter<Web::CSS::Selector> : Formatter<StringView> {
+    ErrorOr<void> format(FormatBuilder& builder, Web::CSS::Selector const& selector)
+    {
+        return Formatter<StringView>::format(builder, selector.serialize());
+    }
+};
+}
```

Comment:
Quite a nitpick, but personally I'd rather see the struct declaration fully qualified, rather than add an enclosing namespace. I don't have a strong argument for it other than it vaguely sort of feels better than opening the AK namespace :grinning: 

```suggestion
template<>
struct AK::Formatter<Web::CSS::Selector> : Formatter<StringView> {
    ErrorOr<void> format(FormatBuilder& builder, Web::CSS::Selector const& selector)
    {
        return Formatter<StringView>::format(builder, selector.serialize());
    }
};
```

Reply:
Oooh, I didn't know you could do that! Opening the namespace is what I've seen used and what I've copied from in the past, but this is a lot nicer, thanks!